src/Raven.Server/Rachis/Leader.cs

```diff
@@ -783,12 +793,15 @@ private void EmptyQueue()
                 {
                     if (_running.IsRaised() == false)
                     {
-                        e = new NotLeadingException("We are no longer the leader, this leader is disposed", e);
+                        e = new NotLeadingException(leaderDisposedMessage, e);
                     }
                     foreach (var tcs in list)
                     {
                         tcs.TrySetException(e);
                     }
+
+                    _appendToRaftLogException = e;
+                    _errorOccurred.Set();
```

Comment:
Not that happy about this, to be honest.
What if we do a `TaskCompletionSource<Exxception>`, instead? Then instead of having separate values, we have just one.
We can still wait on the `WaitHandle` on the Task, instead.

Reply:
I agree it would be much cleaner.
let's do it like this `((IAsyncResult)_error.Task).AsyncWaitHandle`

Reply:
fixed